crates/sui-tool/src/replay.rs

```diff
@@ -1101,7 +1139,13 @@ impl ResourceResolver for LocalExec {
         address: &AccountAddress,
         typ: &StructTag,
     ) -> Result<Option<Vec<u8>>, Self::Error> {
-        let object: Object = self.get_or_download_object(&ObjectID::from(*address))?;
+        let object: Object = match self
+            .get_or_download_object(&ObjectID::from(*address))
+            .map_err(|e| SuiError::GenericStorageError(e.to_string()))?
```

Comment:
Feels a bit strange to use `map_err` and a match imo, could just expand this out into all 3 cases, `Err(_) | Ok(None) | Ok(Some(_))`